src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,205 @@
+                    final ArrayList<String> arrayList = new ArrayList<>();
+                    arrayList.add(checkName);
+                    checksNamesMap.put(packageName, arrayList);
+                }
+                else {
+                    checksNamesMap.get(packageName).add(checkName);
+                }
+            }
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest();
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+        }
+        final Map<String, ArrayList<String>> checksNamesMap = getXdocsMap();
+        for (List<String> sub : checkHandler.checkNamesList) {
+            final String moduleName = sub.get(0);
+            final String checkNameInAttribute = sub.get(1);
+            final String checkNameInText = sub.get(2);
+            final String checkNameInconsistentErrorMsg = String.format(Locale.ROOT,
+                    "Check with name '%s' in attribute "
+                            + "is not consistent with check name in text in file '%s'",
+                    checkNameInAttribute, AVAILABLE_CHECKS_PATH);
+            assertWithMessage(checkNameInconsistentErrorMsg)
+                    .that(checkNameInText).matches(checkNameInAttribute);
+            final String checkNameModuleErrorMsg = String.format(Locale.ROOT,
+                    "Check with name '%s' is not in '%s' module",
+                    checkNameInAttribute, moduleName);
+            if (checkNameInAttribute.equals(COMMENTS_INDENTATION)
+                    || checkNameInAttribute.equals(INDENTATION)) {
+                assertWithMessage(checkNameModuleErrorMsg).that(moduleName).matches(MISC);
+            }
+            else if (checkNameInAttribute.equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertWithMessage(checkNameModuleErrorMsg).that(moduleName).matches(ANNOTATION);
+            }
+            else {
+                final Iterable<String> moduleFileNames = checksNamesMap.get(moduleName);
+                final String moduleNameErrorMsg = String.format(Locale.ROOT,
+                        "module name: '%s' is not exist in '%s'", moduleName, PACKAGE_NAME);
+                assertWithMessage(moduleNameErrorMsg).that(moduleFileNames).isNotNull();
+                boolean match = false;
+                final String checkNameWithSuffix = checkNameInAttribute + SUFFIX_CHECK;
+                for (String filename : moduleFileNames) {
+                    if (checkNameWithSuffix.equals(filename)) {
+                        match = true;
+                        break;
+                    }
+                }
+                assertWithMessage(checkNameModuleErrorMsg).that(match).isTrue();
```

Comment:
Why not `moduleFileNames.contains(checkNameWithSuffix)`?

Reply:
done